extensions/vtk/src/toolbarModule.js

```diff
@@ -5,119 +5,135 @@ const TOOLBAR_BUTTON_TYPES = {
     },
-    operationButtons: [
-      {
-        id: 'setBlendModeToMaximumIntensity',
-        label: 'MIP',
-        type: TOOLBAR_BUTTON_TYPES.SET_TOOL_ACTIVE,
-        commandName: 'setBlendModeToMaximumIntensity',
-        commandOptions: {},
-      },
-      {
-        id: 'setBlendModeToMinimumIntensity',
-        label: 'MinIP',
-        type: TOOLBAR_BUTTON_TYPES.SET_TOOL_ACTIVE,
-        commandName: 'setBlendModeToMinimumIntensity',
+    {
+      id: 'setBlendModeToMaximumIntensity',
+      label: 'Enable MIP',
+      icon: 'soft-tissue',
+      //
+      type: TOOLBAR_BUTTON_TYPES.COMMAND,
+      commandName: 'setBlendModeToMaximumIntensity',
+      commandOptions: {},
+    },
+
+    {
+      id: 'increaseSlabThickness',
+      label: 'Increase Slab Thickness',
+      icon: 'caret-up',
+      //
+      type: TOOLBAR_BUTTON_TYPES.COMMAND,
+      commandName: 'increaseSlabThickness',
+      commandOptions: {},
+    },
+    {
+      id: 'decreaseSlabThickness',
+      label: 'Decrease Slab Thickness',
+      icon: 'caret-down',
+      //
+      type: TOOLBAR_BUTTON_TYPES.COMMAND,
+      commandName: 'decreaseSlabThickness',
+      commandOptions: {},
+    },
+    */
+    {
+      id: 'changeSlabThickness',
+      label: 'Slab Thickness',
+      icon: 'soft-tissue',
+      CustomComponent: SlabThicknessToolbarComponent,
+      commandName: 'setSlabThickness',
+      actionButton: {
+        id: 'setSlabThickness',
+        label: 'slider',
+        type: TOOLBAR_BUTTON_TYPES.COMMAND,
+        commandName: 'setSlabThickness',
         commandOptions: {},
       },
-      {
-        id: 'setBlendModeToAverageIntensity',
-        label: 'AvgIP',
+      deactivateButton: {
+        id: 'setBlendModeToComposite',
         type: TOOLBAR_BUTTON_TYPES.SET_TOOL_ACTIVE,
-        commandName: 'setBlendModeToAverageIntensity',
+        commandName: 'setBlendModeToComposite',
         commandOptions: {},
       },
-    ],
-  },
-];
-
-export default {
-  definitions,
-  defaultContext: 'ACTIVE_VIEWPORT::VTK',
+      operationButtons: [
+        {
+          id: 'setBlendModeToMaximumIntensity',
+          label: 'MIP',
+          type: TOOLBAR_BUTTON_TYPES.SET_TOOL_ACTIVE,
+          commandName: 'setBlendModeToMaximumIntensity',
+          commandOptions: {},
+        },
+        {
+          id: 'setBlendModeToMinimumIntensity',
+          label: 'MinIP',
+          type: TOOLBAR_BUTTON_TYPES.SET_TOOL_ACTIVE,
+          commandName: 'setBlendModeToMinimumIntensity',
+          commandOptions: {},
+        },
+        {
+          id: 'setBlendModeToAverageIntensity',
+          label: 'AvgIP',
+          type: TOOLBAR_BUTTON_TYPES.SET_TOOL_ACTIVE,
+          commandName: 'setBlendModeToAverageIntensity',
+          commandOptions: {},
+        },
+      ],
+    },
+    {
+      id: 'Exit2DMPR',
+      label: 'Exit 2D MPR',
+      icon: 'times',
+      //
+      type: TOOLBAR_BUTTON_TYPES.COMMAND,
+      commandName: 'setCornerstoneLayout',
+    }
```

Comment:
I think the only other way to do this, would be to add this button in `cornerstone`'s extension, but set the context to `ACTIVE_VIEWPORT::VTK`. So the button would only show if VTK viewport was added; bypassing the whole "was this extension registered" conundrum

Reply:
Done. 👍